catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java

```diff
@@ -76,14 +81,42 @@ public String getId() {
 
     @Override
     public <T> boolean canHandle(T subject) {
+        boolean canHandle = false;
+
         if (subject instanceof Metacard) {
             Metacard metacard = (Metacard) subject;
-            String contentTypeName = metacard.getContentTypeName();
-            if (NITF_CONTENT_TYPE.equals(contentTypeName) && metacard.getLocation() != null) {
-                return true;
+
+            boolean isImageNitf = IMAGE_NITF_METACARD_TYPE
+                    .equals(metacard.getMetacardType().getName());
+            boolean hasLocation = metacard.getLocation() != null;
+            boolean hasDerivedImage = hasOriginalDerivedResource(metacard); // original is required for chipping
+
+            canHandle = isImageNitf && hasLocation && hasDerivedImage;
+        }
+
+        return canHandle;
+    }
+
+    private boolean hasOriginalDerivedResource(Metacard metacard) {
```

Comment:
This method has high cyclomatic complexity. You might consider this functional approach:
```java
    private boolean hasOriginalDerivedResource(Metacard metacard) {
        Attribute attribute = metacard.getAttribute(Core.DERIVED_RESOURCE_URI);

        return Stream.of(attribute)
                .filter(Objects::nonNull)
                .flatMap(a -> a.getValues()
                        .stream())
                .filter(String.class::isInstance)
                .map(String.class::cast)
                .anyMatch(this::hasOriginalQual);
    }

    private boolean hasOriginalQual(String uriString) {
        try {
            URI derivedResourceUri = new URI(uriString);

            // find the #original URI fragment
            if (ORIGINAL_QUALIFIER.equals(derivedResourceUri.getFragment())) {
                return true;
            }
        } catch (URISyntaxException use) {
            LOGGER.debug("Could not parse URI string [{}]", uriString);
        }
        return false;
    }
```